libs/providers/flagd-web/src/lib/flagd-web-provider.ts

```diff
@@ -0,0 +1,162 @@
+  ): Promise<ResolutionDetails<boolean>> {
+      return this.promiseClient.resolveBoolean({
+        flagKey,
+        context: Struct.fromJsonString(JSON.stringify(transformedContext)),
+      }).then((res) => {
+        return {
+          value: res.value,
+          reason: res.reason,
+          variant: res.variant,
+        }
+      }).catch((err: unknown) => {
+        return {
+          reason: StandardResolutionReasons.ERROR,
+          errorCode: ErrorResponse(err),
+          value: defaultValue,
+        };
+      })
+  }
+
+  resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    transformedContext: EvaluationContext
+  ): Promise<ResolutionDetails<string>> {
+    return this.promiseClient.resolveString({
+      flagKey,
+      context: Struct.fromJsonString(JSON.stringify(transformedContext)),
+    }).then((res) => {
+      return {
+        value: res.value,
+        reason: res.reason,
+        variant: res.variant,
+      }
+    }).catch((err: unknown) => {
+      return {
+        reason: StandardResolutionReasons.ERROR,
+        errorCode: ErrorResponse(err),
+        value: defaultValue,
+      };
+    })
+  }
+
+  resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    transformedContext: EvaluationContext
+  ): Promise<ResolutionDetails<number>> {
+    return this.promiseClient.resolveFloat({
+      flagKey,
+      context: Struct.fromJsonString(JSON.stringify(transformedContext)),
+    }).then((res) => {
+      return {
+        value: res.value,
+        reason: res.reason,
+        variant: res.variant,
+      }
+    }).catch((err: unknown) => {
+      return {
+        reason: StandardResolutionReasons.ERROR,
+        errorCode: ErrorResponse(err),
+        value: defaultValue,
+      };
+    })
+  }
+
+  resolveObjectEvaluation<U extends JsonValue>(
+    flagKey: string,
+    defaultValue: U,
+    transformedContext: EvaluationContext
+  ): Promise<ResolutionDetails<U>> {
+    return this.promiseClient.resolveObject({
+      flagKey,
+      context: Struct.fromJsonString(JSON.stringify(transformedContext)),
+    }).then((res) => {
+      if (res.value) {
+        return {
+          value: JSON.parse(res.value.toJsonString()) as U,
+          reason: res.reason,
+          variant: res.variant,
+        }
+      }
+      return {
+        value: defaultValue,
+        reason: res.reason,
+        variant: res.variant,
+      }
+    }).catch((err: unknown) => {
+      return {
+        reason: StandardResolutionReasons.ERROR,
+        errorCode: ErrorResponse(err),
+        value: defaultValue,
+      };
+    })
+  }
+}
+
+function ErrorResponse(err: unknown): string {
+  err as Partial<ConnectError>
+  switch ((err as Partial<ConnectError>).code) {
+    case Code.NotFound:
+      return ErrorCode.FLAG_NOT_FOUND
+    case Code.InvalidArgument:
+      return ErrorCode.TYPE_MISMATCH
+    case Code.Unavailable:
+        return ERROR_DISABLED
```

Comment:
I think we may be mixing reasons/error semantics here.

I don't think disabled flag actually constitutes an error. This is evidenced by the fact it's not an enumerate error code, and it's really not an unexpected condition. A flag being disabled is something somebody chose to do, I don't think that logically it's an error condition. It's a `reason`.

cc @beeme1mr